### PR TITLE
rename table field

### PR DESCRIPTION
**Pull request by Ashleyhx**

### CODE REVIEW

**Comment by Ashleyhx** [comment]
Able to rename table from SchemaView, graph, and settingsView will change accordingly.
image
image
image

File: src/components/SchemaView/SchemaSidebarEditView.vue [hunk 1]
```diff
@@ -234,6 +248,23 @@ export default {
     getColor(label) {
       return this.settingsStore.colorForLabel(label);
     },
+    enterRenameMode() {
+      this.renameTable = true;
+    },
+    saveNewTableName() {
+      if (this.currLabel === this.label) {
```
Comment (mewim) [inline src/components/SchemaView/SchemaSidebarEditView.vue:255]: This should not be required either. If the name does not change, you do not need to show the buttons (which means this is not possible).

File: src/components/SchemaView/SchemaSidebarEditView.vue [hunk 2]
```diff
@@ -9,18 +9,24 @@
...
+          @click="enterRenameMode"
           :style="{
             backgroundColor: ` ${getColor(label)} !important`,
             color: isNode ? '#ffffff' : '#000000',
           }"
-        >
-          {{ label }}
-        </span>
+        />
+        <button v-if="renameTable" @click="saveNewTableName" class="btn btn-sm btn-outline-primary">
```
Comment (mewim) [inline src/components/SchemaView/SchemaSidebarEditView.vue:24]: The buttons can simply be shown / hidden by comparing if `currLabel === label"

File: src/components/SchemaView/SchemaViewMain.vue [hunk 3]
```diff
@@ -673,6 +672,16 @@ export default {
       this.clickedLabel = newLabel;
     },
 
+    renameTable(oldLabel, newLabel) {
+      this.$emit("renameTable", oldLabel, newLabel);
+      this.clickedLabel = newLabel;
+      this.settingsStore.renameNodeTable(oldLabel, newLabel);
```
Comment (mewim) [inline src/components/SchemaView/SchemaViewMain.vue:678]: This should not be done. Instead, you should follow the approach of `updatePlaceholderNodeTableLabel`.

File: src/components/SchemaView/SchemaSidebarEditView.vue [hunk 4]
```diff
@@ -9,18 +9,24 @@
           @click="$emit('back')"
         >
           <i class="fa-solid fa-long-arrow-left"></i>
-          Back
         </button>
-        &nbsp; Editing {{ isNode ? "Node" : "Rel" }} Table: &nbsp;
-        <span
-          class="badge bg-primary"
+        &nbsp; Table: &nbsp;
```
Comment (mewim) [inline src/components/SchemaView/SchemaSidebarEditView.vue:13]: This label can be removed I think

File: kuzu
Comment (mewim) [inline kuzu:1]: Discard submodule change

**Comment by mewim** [comment]
> Able to rename table from SchemaView, graph, and settingsView will change accordingly. image image image

The behavior as shown in 1st screenshot is incorrect. The table input has been changed to `cc` yet the diagram is showing `City` still.

Also, you should improve the visual style:
1. Align the buttons and the input
2. Take full space
3. Adjust button spacing

**Comment by mewim** [comment]
Also, maybe I missed something? You used `updatePlaceholderRelTable` but did not mark any table as a placeholder to begin with, so this might work just as a coincidence.

**Review by mewim — CHANGES_REQUESTED** [review]
See my comments

File: src/utils/Constants.js [hunk 5]
```diff
@@ -68,4 +68,4 @@ export const MODES = {
 };
 
 export const PLACEHOLDER_NODE_TABLE = "__placeholder_node_table__";
-export const PLACEHOLDER_REL_TABLE = "__placeholder_rel_table__";
+export const PLACEHOLDER_REL_TABLE = "__placeholder_rel_table__";
```
Comment (mewim) [inline src/utils/Constants.js:71]: You should not remove the new line here.

File: src/utils/DataDefinitionLanguage.js [hunk 6]
```diff
@@ -126,6 +126,18 @@ class DataDefinitionLanguage {
     result += ");";
     return result;
   }
+
+  saveRenamedNodeTable(oldName, newName) {
```
Comment (mewim) [inline src/utils/DataDefinitionLanguage.js:130]: The two functions below should be merged. Also, it should just be `renameTable`.

File: src/components/SchemaView/SchemaSidebarEditView.vue [hunk 7]
```diff
@@ -234,6 +250,20 @@ export default {
     getColor(label) {
       return this.settingsStore.colorForLabel(label);
     },
+    saveNewTableName() {
```
[the remark this answers is not in the document]
Comment (mewim) [reply src/components/SchemaView/SchemaSidebarEditView.vue:253]: Also, the event name can be simplified as `renameNodeTable` and `renamedRelTable `

File: src/components/SchemaView/SchemaViewMain.vue [hunk 8]
```diff
@@ -655,10 +657,39 @@ export default {
       this.reloadSchema();
     },
 
+
+    enterRenameNodeTableMode(newTableName) {
```
Comment (mewim) [inline src/components/SchemaView/SchemaViewMain.vue:661]: If the code here is not being used, you should remove the code instead of commenting out

**Pull request merged**